src/simsopt/_core/finite_difference.py

```diff
@@ -253,6 +258,7 @@ def _jac(self, x: RealArray = None):
                 x = xs[:, j]
                 mpi.comm_groups.bcast(x, root=0)
                 opt.x = x
+                self.opt.non_dofs = non_dofs
```

Comment:
How is the non_dofs attribute used in the optimizable object? Is there any convention or standardization you have in mind here?

Reply:
I was thinking that it should be an array of doubles. I have no other use cases in mind.

Reply:
See example here
https://github.com/hiddenSymmetries/simsopt/blob/rj/single_stage_PR/examples/3_Advanced/single_stage_optimization_finite_beta.py#L223

Reply:
This smells funny to me, because `non_dofs` is an attribute only of objects created by `make_optimizable()`. Most optimizable objects don't have a `non_dofs` attribute.